From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Helvetica Volubi <suisuroru@blue-millennium.fun>
Date: Tue, 21 Apr 2026 23:47:15 +0800
Subject: [PATCH] Leaf: Remove useless creating stats json bases on player name
 logic

Co-authored by: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
As part of: Leaf (https://github.com/Winds-Studio/Leaf/blob/7f3e240bbe0970683c40279a7a65f0fde47503b6/leaf-server/minecraft-patches/features/0043-Remove-useless-creating-stats-json-bases-on-player-n.patch)
Licensed under: MIT (https://github.com/Winds-Studio/Leaf/blob/7f3e240bbe0970683c40279a7a65f0fde47503b6/licenses/MIT.txt)

diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index 551354f4f5258b2aa323b30167d413e894779f05..b3327f65319f1717e40a3704ab000e6cfeaf9750 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -1295,10 +1295,12 @@ public abstract class PlayerList {
         if (Files.exists(uuidStatsFile)) {
             return uuidStatsFile;
         } else {
-            String playerNameStatsFile = gameProfile.name() + ".json";
-            if (FileUtil.isValidPathSegment(playerNameStatsFile)) {
-                Path playerNameStatsPath = statFolder.resolve(playerNameStatsFile);
-                if (Files.isRegularFile(playerNameStatsPath)) {
+            // Leaf start - Remove useless creating stats json bases on player name logic
+            /*
+            String string = profile.name() + ".json";
+            if (FileUtil.isValidPathSegment(string)) {
+                Path path1 = worldPath.resolve(string);
+                if (Files.isRegularFile(path1)) {
                     try {
                         return Files.move(playerNameStatsPath, uuidStatsFile);
                     } catch (IOException var7) {
@@ -1307,6 +1309,8 @@ public abstract class PlayerList {
                     }
                 }
             }
+            */
+            // Leaf end - Remove useless creating stats json bases on player name logic
 
             return uuidStatsFile;
         }
